src/alerts/flash-refactor.scss

```diff
@@ -0,0 +1,133 @@
+// stylelint-disable selector-max-type
+
+// Flash alert
+
+.Flash {
+  position: relative;
+  display: grid;
+  padding: $spacer-2;
+  color: var(--color-fg-default);
+  background-image: linear-gradient(var(--color-accent-subtle), var(--color-accent-subtle));
+  border: $border-width $border-style var(--color-accent-muted);
+  border-radius: $border-radius;
+  grid-auto-flow: column;
+  grid-template-areas: 'visual message actions close';
+  grid-template-columns: min-content 1fr min-content min-content;
+  grid-template-rows: min-content;
+
+  // `sm` breakpoint variantion
+  @media (max-width: #{map-get($breakpoints, 'sm') - 0.02px}) {
+    grid-template-areas:
+      'visual message close'
+      '  .    actions actions';
+    grid-template-columns: min-content 1fr min-content;
+    grid-template-rows: min-content min-content;
+
+    .Flash-actions {
+      margin: $spacer-2 0 0 $spacer-2;
+    }
+  }
+
+  // Elements
+
+  .Flash-visual {
+    display: grid;
+    // stylelint-disable-next-line primer/spacing
+    padding: calc((#{$spacer-5} - 20px) / 2) $spacer-2;
+    grid-area: visual;
+    align-self: start;
+
+    > .octicon {
+      margin-block: calc((#{$spacer-1}) / 2);
+    }
+
+    > * {
+      align-self: center;
+    }
+  }
+
+  .Flash-message {
+    // stylelint-disable-next-line primer/spacing
+    padding: calc((#{$spacer-5} - 20px) / 2) $spacer-2;
+    grid-area: message;
+    align-self: center;
+
+    p:last-child {
+      margin-bottom: 0;
+    }
+  }
+
+  .Flash-actions {
+    grid-area: actions;
+
+    &:last-child {
+      align-self: end;
```

Comment:
This might be out of scope (I'm not sure if you're working from existing designs) but the trailing action alignment stands out to me. I _feel_ like I want it to align with the text. What do you think?

<img width="1083" alt="flash alert with button aligned with text" src="https://user-images.githubusercontent.com/18661030/156682123-f915524d-4696-4b3b-8028-4fabd19442ed.png">

Reply:
Or keep at `start` so it aligns with the icon regardless how much content there is:

![image](https://user-images.githubusercontent.com/378023/156725900-d591deaa-9965-4ebc-8c85-f954de60b768.png)

Reply:
My line of thought was that, for banners with multiple lines and no dismiss button, the call-to-action should be aligned to the end of the message, like: "read the message before looking at the button".

![Screen Shot 2022-03-14 at 15 45 19](https://user-images.githubusercontent.com/293280/158273020-f5dca274-6f38-43bd-a75d-ced96e6c50cb.png)

If the banner does have a dismiss button, though, I'm still keeping it aligned to the top. For some reason I was imagining that banners that have a dismiss button have call-to-actions that are not as important, since they can be dismissed 🤔 .

![Screen Shot 2022-03-14 at 15 45 06](https://user-images.githubusercontent.com/293280/158272993-cd25e90b-baa3-4544-8f62-f39269ecc4a4.png)

Seeing the banners above with the different alignments can definitely seen as jarring, but these usually don't appear next to each other anyways.

@langermank @simurai Do you think that's too weird? 😆

Reply:
> the call-to-action should be aligned to the end of the message, like: "read the message before looking at the button".

I see. Yeah, I think that makes sense. Let's give it a try.

Reply:
I think it's okay to allow for two different action button positions 👍